### PR TITLE
fix(jellyfish-api-core): update listGovProposals rpc to accept cycle and pagination

### PR DESCRIPTION
**Pull request by kyleleow**
#### What this PR does / why we need it:
- To update `listGovProposals` to accept `cycle` and `pagination` added in https://github.com/DeFiCh/ain/pull/1627
- To add more e2e related to `pagination`

#### Which issue(s) does this PR fixes?:

Fixes #

#### Additional comments?:
- Pending https://github.com/DeFiCh/ain/pull/1689 to fix an issue where `start` is ignored when `status` is not passed OR when `cycle` is passed
- However, this should not be a blocker for now

### CODE REVIEW

File: packages/jellyfish-api-core/src/category/governance.ts [hunk 1]
```diff
@@ -101,13 +101,15 @@ export class Governance {
    * @param {Object} options List proposal filter options
    * @param {ListProposalsType} [options.type=ListProposalsType.ALL] type of proposals
    * @param {ListProposalsStatus} [options.status=ListProposalsStatus.ALL] status of proposals
+   * @param {number} [options.cycle=0] payout cycle of proposals
```
Comment (delphk) [inline packages/jellyfish-api-core/src/category/governance.ts]:
```suggestion
   * @param {number} [options.cycle=0] cycle: 0 (show all), cycle: N (show cycle N), cycle: -1 (show previous cycle)
```
^ suggestion to make the cycle param more informative

File: docs/node/CATEGORIES/12-governance.md [hunk 2]
```diff
@@ -112,10 +112,20 @@ Returns list of proposals.
...
+
+interface ListProposalsOptions {
+  type?: ListProposalsType
+  status?: ListProposalsStatus
+  cycle?: number
+  pagination?: ListProposalsPagination
+}
+
+interface ListProposalsPagination {
+  start?: string
```
Comment (delphk) [inline docs/node/CATEGORIES/12-governance.md:126]: noticed some discrepancy between the expected type for `start` between here and [ain](https://github.com/DeFiCh/ain/blob/master/src/masternodes/rpc_proposals.cpp#L803-L807), but seems like the wrong type is on `ain`'s side.
 cc: @shohamc1

Comment (kyleleow) [reply docs/node/CATEGORIES/12-governance.md:126]: I have informed the blockchain team previously and they are awared about the type error.

File: packages/jellyfish-api-core/__tests__/category/governance/listGovProposals.test.ts [hunk 3]
```diff
@@ -108,6 +120,234 @@ describe('Governance', () => {
...
+      pagination: {
+        including_start: false,
+        limit: 2
+      }
+    })
+    expect(proposals).toStrictEqual(allProposals.slice(1, 3))
+    expect(proposals.length).toStrictEqual(2)
+  })
+
+  // TODO: remove `skip` after blockchain fixes bug where `start` is not considered when non-all status is not passed
```
Comment (delphk) [inline packages/jellyfish-api-core/__tests__/category/governance/listGovProposals.test.ts:256]: just wondering if there is an existing PR or issue for this bug on ain?

Comment (shohamc1) [reply packages/jellyfish-api-core/__tests__/category/governance/listGovProposals.test.ts:256]: It's being worked on ATM, but no PR yet.

**Review by delphk — APPROVED** [review]
looks good overall pending bug fix for skipped tests

File: packages/jellyfish-api-core/src/category/governance.ts (hunk 1 above)
Comment (kyleleow) [reply packages/jellyfish-api-core/src/category/governance.ts]: updated

**Comment by fuxingloh** [comment]
> looks good overall pending bug fix for skipped tests

I assume we should wait until upstream addresses it?

File: packages/jellyfish-api-core/__tests__/category/governance/listGovProposals.test.ts (hunk 3 above)
Comment (kyleleow) [reply packages/jellyfish-api-core/__tests__/category/governance/listGovProposals.test.ts:256]: PR created yesterday https://github.com/DeFiCh/ain/pull/1689

**Review by fuxingloh — APPROVED** [review]

**Pull request merged**